packages/headless/src/features/search/search-actions.ts

```diff
@@ -50,6 +50,10 @@ export interface SearchAction {
   actionCause: string;
 }
 
+export const nullActionCause = (): SearchAction => ({
+  actionCause: null as unknown as string,
+});
```

Comment:
This was way easier to do than fixing the type in the entirety of headless. I tried changing actionCause to string | null and it was breaking everything. Some legacy stuff is using next stuff and some next stuff is using legacy. Simply casting is preferable.